src/java.base/share/classes/java/net/ServerSocket.java

```diff
@@ -299,7 +299,7 @@ private SocketImpl getImpl() throws SocketException {
                     } catch (SocketException e) {
                         throw e;
                     } catch (IOException e) {
-                        throw new SocketException(e.getMessage());
+                        throw new SocketException(e.getMessage(), e);
```

Comment:
This one is probably okay although most of the stack trace of the cause will be identical to that of the SocketException so it might be confusing in logs. The equivalent in Socket might be a bit more useful as there are several SocketImpls used for client sockets.